File: qna/src/main/java/com/example/rag/qna/domain/cache/Cache.java
package com.example.rag.qna.domain.cache;

import core.qna.CacheSystem;
import core.rag.Question;
import core.rag.QuestionTitleDTO;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
class Cache implements CacheSystem {
    private final CategoryQuestionTitleCache questionTitleCache;
    private final QuestionCache questionCache;

    @Override
    public QuestionTitleDTO[] getRecentQuestionTitles(String category, int startNum, int endNum) {
        return questionTitleCache.getRecentQuestionTitles(category, startNum, endNum);
    }

    @Override
    public void putRecentQuestionTitle(String category, QuestionTitleDTO questionTitle) {
        questionTitleCache.putRecentQuestionTitle(category, questionTitle);
    }

    @Override
    public Question getRecentQuestion(String questionId) {
        return questionCache.getRecentQuestion(questionId);
    }

    @Override
    public void putRecentQuestion(String questionId, Question question) {
        questionCache.putQuestion(questionId, question);
    }
}

File: cache/src/main/kotlin/com/example/rag/qna/domain/cache/Cache.kt
package com.example.rag.qna.domain.cache

import core.qna.CacheSystem
import core.rag.Question
import core.rag.QuestionTitle
import org.springframework.stereotype.Component

@Component
class Cache(private val questionTitleCache: CategoryQuestionTitleCache, private val questionCache: QuestionCache): CacheSystem {

    override fun getRecentQuestionTitles(category: String, offset: Int, limit: Int): List<QuestionTitle?>
        = questionTitleCache.getRecentQuestionTitles(category, offset, limit)

    override fun putRecentQuestionTitle(category: String, questionTitle: QuestionTitle) {
        questionTitleCache.putRecentQuestionTitle(category, questionTitle)
    }

    override fun getRecentQuestion(questionId: String): Question?
        = questionCache.getRecentQuestion(questionId)

    override fun putRecentQuestion(questionId: String, question: Question) {
        questionCache.putQuestion(questionId, question)
    }
}